File: Selenium2PlaywrightConverterWithLocalLLM/tools/check_ollama.py
import requests
import json
import sys

def check_ollama(model_name="codellama"):
    base_url = "http://localhost:11434"
    
    # 1. Check if Ollama is running
    try:
        print(f"Checking connection to {base_url}...")
        requests.get(base_url)
        print("✅ Ollama is reachable.")
    except requests.exceptions.ConnectionError:
        print("❌ Ollama is NOT running or not reachable on port 11434.")
        return False

    # 2. Check if model exists
    print(f"Checking for model: {model_name}...")
    try:
        resp = requests.get(f"{base_url}/api/tags")
        models = resp.json().get('models', [])
        found = any(m['name'].startswith(model_name) for m in models)
        
        if found:
            print(f"✅ Model '{model_name}' found.")
        else:
            print(f"⚠️ Model '{model_name}' NOT found in local library.")
            print("Running 'ollama pull codellama' might be needed.")
            # We won't auto-pull as it might be large, just warn.
            
    except Exception as e:
        print(f"❌ Error listing models: {e}")
        return False

    # 3. Handshake (Simple Generation)
    print("Attempting handshake (simple generation)...")
    payload = {
        "model": model_name,
        "prompt": "Write a hello world in python",
        "stream": False
    }
    
    try:
        res = requests.post(f"{base_url}/api/generate", json=payload)
        if res.status_code == 200:
            print("✅ Handshake successful! Response received.")
            print(f"Sample: {res.json().get('response', '')[:50]}...")
            return True
        else:
            print(f"❌ Handshake failed: {res.status_code} - {res.text}")
            return False
    except Exception as e:
        print(f"❌ Handshake error: {e}")
        return False

if __name__ == "__main__":
    success = check_ollama()
    if not success:
        sys.exit(1)

File: Selenium2PlaywrightConverterWithLocalLLM/tools/check_ollama.js
const http = require('http');

const checkOllama = async () => {
    const modelName = 'codellama';
    const options = {
        hostname: 'localhost',
        port: 11434,
        path: '/api/tags',
        method: 'GET',
    };

    console.log(`Checking connection to http://${options.hostname}:${options.port}...`);

    const req = http.request(options, (res) => {
        let data = '';

        res.on('data', (chunk) => {
            data += chunk;
        });

        res.on('end', () => {
            if (res.statusCode === 200) {
                console.log('✅ Ollama is reachable.');
                try {
                    const models = JSON.parse(data).models || [];
                    const found = models.some((m) => m.name.startsWith(modelName));

                    if (found) {
                        console.log(`✅ Model '${modelName}' found.`);
                        performHandshake();
                    } else {
                        console.log(`⚠️ Model '${modelName}' NOT found in local library.`);
                        console.log(`Please run: ollama pull ${modelName}`);
                        process.exit(1);
                    }
                } catch (e) {
                    console.error('❌ Error parsing response:', e.message);
                    process.exit(1);
                }
            } else {
                console.log(`❌ Ollama returned status: ${res.statusCode}`);
                process.exit(1);
            }
        });
    });

    req.on('error', (e) => {
        console.error(`❌ Ollama is NOT reachable: ${e.message}`);
        process.exit(1);
    });

    req.end();
};

const performHandshake = () => {
    console.log('Attempting handshake (simple generation)...');
    const postData = JSON.stringify({
        model: 'codellama',
        prompt: 'Write a hello world in javascript',
        stream: false,
    });

    const options = {
        hostname: 'localhost',
        port: 11434,
        path: '/api/generate',
        method: 'POST',
        headers: {
            'Content-Type': 'application/json',
            'Content-Length': postData.length,
        },
    };

    const req = http.request(options, (res) => {
        let data = '';

        res.on('data', (chunk) => {
            data += chunk;
        });

        res.on('end', () => {
            if (res.statusCode === 200) {
                console.log('✅ Handshake successful! Response received.');
                try {
                    const json = JSON.parse(data);
                    console.log(`Sample: ${json.response.substring(0, 50)}...`);
                } catch (e) {
                    console.log('Raw response received (parsing failed but connection works).');
                }
            } else {
                console.log(`❌ Handshake failed: ${res.statusCode}`);
                process.exit(1);
            }
        });
    });

    req.on('error', (e) => {
        console.error(`❌ Handshake error: ${e.message}`);
        process.exit(1);
    });

    req.write(postData);
    req.end();
};

checkOllama();
